elastic/kibana
# [Reporting] Add deprecation notice to the upgrade assistant

## Summary

Continuation of https://github.com/elastic/kibana/pull/100130 and fix https://github.com/elastic/kibana/issues/81544

This contribution adds an integration with the deprecation service. The deprecation flags the new reporting policy and provides a "quick resolution" in the upgrade assistant which uses the new "PUT <base-path>/deprecations" endpoint.

## How to test

1. Start Kibana and ES, on basic license level
2. Make the following request against the ES:
```bash
curl --request PUT \
  --url 'http://localhost:9200/.reporting-*/_settings' \
  --header 'Authorization: Basic ZWxhc3RpYzpjaGFuZ2VtZQ==' \
  --header 'Content-Type: application/json' \
  --data '{
	"settings": {
		"index.lifecycle.name": null
	}
}'
```
3. Go the Upgrade Assistant (in code) and do the following:

```diff
diff --git i/x-pack/plugins/upgrade_assistant/public/plugin.ts w/x-pack/plugins/upgrade_assistant/public/plugin.ts
index 4cffd40faf3..701eefc33b6 100644
--- i/x-pack/plugins/upgrade_assistant/public/plugin.ts
+++ w/x-pack/plugins/upgrade_assistant/public/plugin.ts
@@ -58,7 +58,7 @@ export class UpgradeAssistantUIPlugin implements Plugin {
           coreSetup,
           params,
           kibanaVersionInfo,
-          readonly
+          false
         );
 
         return () => {

```
4. Go the upgrade assistant in the UI and see the deprecation warning there

## Screenshots

Please note: the current Upgrade Assistant UI is subject to a lot of change prior to 8.0.

<img width="1655" alt="Screenshot 2021-07-05 at 16 06 24" src="https://user-images.githubusercontent.com/8155004/124483801-081a0080-ddab-11eb-901a-80872fa950b9.png">


### Checklist

Delete any items that are not applicable to this PR.
